dotnet/src/Connectors/Connectors.HuggingFace/Core/IHuggingFaceClient.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Connectors.HuggingFace.Core;
+
+internal interface IHuggingFaceClient
+{
+    /// <summary>
+    /// Generates text based on the given prompt asynchronously.
+    /// </summary>
+    /// <param name="prompt">The prompt for generating text content.</param>
+    /// <param name="executionSettings">The prompt execution settings (optional).</param>
+    /// <param name="cancellationToken">The cancellation token (optional).</param>
+    /// <returns>A list of text content generated based on the prompt.</returns>
+    Task<IReadOnlyList<TextContent>> GenerateTextAsync(
+        string prompt,
+        PromptExecutionSettings executionSettings,
+        CancellationToken cancellationToken);
+
+    /// <summary>
+    /// Streams the generated text content asynchronously.
+    /// </summary>
+    /// <param name="prompt">The prompt for generating text content.</param>
+    /// <param name="executionSettings">The prompt execution settings (optional).</param>
+    /// <param name="cancellationToken">The cancellation token (optional).</param>
+    /// <returns>An asynchronous enumerable of <see cref="StreamingTextContent"/> streaming text contents.</returns>
+    IAsyncEnumerable<StreamingTextContent> StreamGenerateTextAsync(
+        string prompt,
+        PromptExecutionSettings? executionSettings = null,
+        CancellationToken cancellationToken = default);
```

Comment:
similar comment about having default value for cancellation tokens for public methods of the internal implementation class.